cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+
+1. Deb creates the resource provider and the resource type with Radius
+
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    ```
+    `Mycompany.Messaging.yaml` is the schema definition file that contains the resource type definition for the messaging service Plaid. See the feature spec that covers the [authoring experience for the schema definition](https://github.com/radius-project/design-notes/blob/67dd03056d22bcae2ee9743ffb308c8ac7f012bd/architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md)
+
+2. Deb creates the resource instance for the messaging service Plaid
+
+    ```bash
+    rad resource create Mycompany.Messaging/Plaid myplaid -f myplaid.json
+    ```
+    The resource `myplaid` of type `Mycompany.Messaging/plaidResource`is created and available for use.
+
+4. Deb lists the resource providers and resource types available in Radius
+
+    ```bash
+    rad resource-provider list
+    ```
+    The list of resource providers are displayed.
+
+    ```bash
+    rad resource-type list --resource-provider Mycompany.Messaging
+    ```
+    The list of resource types under the resource provider `Mycompany.Messaging` are displayed.
+
+5. Deb makes changes to the resource type definition of `plaidResource` and adds to the resource provider
+
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    ```
+
+6. Deb wants to know the details of resource provider
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    ```
+    The details of the resource provider `Mycompany.Messaging` are displayed.
+
+7. Deb deletes a resource-type Plaid from the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-type delete Mycompany.Messaging/plaidResource
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource type plaidResource from the resource provider Mycompany.Messaging? (yes/no)
+    ```
+    The resource type `Plaid` is deleted from the resource provider `Mycompany.Messaging`
+
+8. Deb deletes the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource provider Mycompany.Messaging? (yes/no)
+    ```
+
+## Finalized set of commands
+
+### rad resource-provider
+
+#### 1. rad resource-provider create
+
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Creating a resource provider involves creating new resource types that can be used in your applications. Inputs can be passed in as a file or as flags.
+
+Create a resource provider from a file 
+    
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
```

Comment:
Would it make sense to support URLs here as well? eg: a link to a GH repo. Perhaps **our** github repo 😁